prdoc/pr_4706.prdoc

```diff
@@ -0,0 +1,17 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: "Rename `assigner_on_demand` pallet to `on_demand`"
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Renames `assigner_on_demand` pallet to `on_demand`
+
+crates: 
+- name: polkadot-runtime-parachains
+  bump: minor
```

Comment:
It is renaming public modules, i think that should be `major`:
![Screenshot 2024-07-22 at 15 08 31](https://github.com/user-attachments/assets/153fee85-8fb8-4c8d-8624-e9390c184301)

Reply:
Done!